xpra/server/auth/keycloak.py

```diff
@@ -158,9 +164,37 @@ def check(self, response_json: bytes) -> bool:
                 log.error("Error: keycloak authentication failed as token state is invalid")
                 return False
 
+            log("keycloak: token is active")
+
             user_info = keycloak_openid.userinfo(access_token)
-            log("userinfo_info: %r", user_info)
-            log("keycloak authentication succeeded: token is active")
+            log("user_info: %r", user_info)
+
+            log("claim_field: %r", self.claim_field)
+            log("auth_groups: %r", self.auth_groups)
+            # if claim_field is defined, check that the auth_groups match the groups_claim condition
+            if self.claim_field:
+                if not self.auth_groups:
+                    log.error("Error: keycloak authentication failed as auth_groups is invalid")
+                    return False
+
+                def get(data_dict, keys):
+                    return get(data_dict[keys.pop(0)], keys) if len(keys) else data_dict
```

Comment:
It's still not clear what that does.
Please include docstrings with an example to make it easier for someone reading this code.